### PR TITLE
Set backends to active or inactive

### PR DESCRIPTION
**Pull request by nolancon**
### Description of your changes
Mark backends as active or inactive instead of deleting from backend store.

Related to #23

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

**Comment by nolancon** [comment]
@mhmxs WDYT... was this something like what you had in mind?

File: internal/backendstore/backend.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+func (be *backend) isBackendActive() bool {
+	be.mu.RLock()
+	defer be.mu.RUnlock()
+
+	return be.active
+}
+
+func (be *backend) setBackendActive(active bool) {
```
Comment (mhmxs) [inline internal/backendstore/backend.go:36]: This function doesn't sets backend to active, but toggles the status. Does it make sense to rename the function?

File: internal/backendstore/backendstore.go [hunk 2]
```diff
@@ -21,21 +21,64 @@ func NewBackendStore() *BackendStore {
 	}
 }
 
+func (b *BackendStore) GetBackendClient(backendName string) *s3.Client {
+	b.mu.RLock()
```
Comment (mhmxs) [inline internal/backendstore/backendstore.go:25]: Does it make sense to merge these functions with `backend` struct to save one lock? In this way we have two locks, just to be programatically correct. We won't use `backend` without these helpers. WDYT?

Comment (nolancon) [reply internal/backendstore/backendstore.go:25]: Could you elaborate on this a bit more please? I'm not sure I understand what this would look like...

Comment (mhmxs) [reply internal/backendstore/backendstore.go:25]: Currently we call `GetBackendClient` then lock `b.mu.RLock()` then call `getBackendClient` then we lock again `be.mu.RLock()`. But we never use `getBackendClient` only from `GetBackendClient` so maybe if we should combine the functionality into one, we should have only one lock. WDYT?

Comment (nolancon) [reply internal/backendstore/backendstore.go:25]: Ah yes, sorry. Makes sense :+1:

**Pull request merged**